Assets/__Scripts/Requirements/VivifyReq.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Beatmap.Base;
+using Beatmap.Info;
+using UnityEngine;
+
+public class VivifyReq : RequirementCheck
+{
+    public override string Name => "Vivify";
+
+    public override RequirementType IsRequiredOrSuggested(InfoDifficulty infoDifficulty, BaseDifficulty map)
+        => MapHasVivifyBundles() || MapHasVivifyEvents(map) ? RequirementType.Requirement : RequirementType.None;
+    
+    private bool MapHasVivifyEvents(BaseDifficulty map) =>
+        map.CustomEvents.Any(ev => vivifyEventTypes.Contains(ev.Type));
+
+    private bool MapHasVivifyBundles() =>
+        BeatSaberSongContainer.Instance.Map.CustomData != null &&
+        BeatSaberSongContainer.Instance.Map.CustomData.HasKey("_assetBundle");
```

Comment:
Looking at Vivify docs, isn't the `_assetBundle` key in the Info and not the difficulty?
```suggestion
        BeatSaberSongContainer.Instance.MapDifficultyInfo.CustomData.HasKey("_assetBundle");
```